### ISSUE TITLE
[Feature Request] Consistent result of serialization no matter createFactory is true or flase

### ISSUE THREAD

**Issue by Perfetto2020** [google/json_serializable.dart#1102]
Currently the result of serialization varies a little with setting `createFactory` true or false.
``` dart
abstract class Equatable {
  @override
  int get hashCode => 0xffff;

}
@JsonSerializable()
class Person extends Equatable {
  final String name;
  final int gender;
  int age;

  bool get nameAwesome => name.startsWith('A');

  Person({required this.name}) : age = 18, gender = 1;
}
```
If set `createFactory` to true(default value) as shown above, the generated `ToJson` will be as follows and looks good:

``` dart
Map<String, dynamic> _$PersonToJson(Person instance) => <String, dynamic>{
      'name': instance.name,
      'age': instance.age,
    };
```
But if specify `createFactory` to `false`:
``` dart
@JsonSerializable(createFactory: false)
class Person extends Equatable {
```
The generated `ToJson` will be as follows:
``` dart
Map<String, dynamic> _$PersonToJson(Person instance) => <String, dynamic>{
      'hashCode': instance.hashCode,
      'name': instance.name,
      'gender': instance.gender,
      'age': instance.age,
      'nameAwesome': instance.nameAwesome,
    };
```
As shown in codes above, 3 more fields are serialized:
`hashCode`: a `getter` inherited from parent;
`nameAwesome`: a `getter` defined itself;
`gender`: a field initialized in initializer list.

I think,
1. `getter`s shouldn't be serialized especially those inherited from parent, unless it's annotated by `JsonKey`; issue [968](https://github.com/google/json_serializable.dart/issues/968), [815](https://github.com/google/json_serializable.dart/issues/815),
2. fields initialized in initializer list should be serialized.

So my expected `ToJson` no matter `createFactory` is true or false should be as follows:
``` dart
Map<String, dynamic> _$PersonToJson(Person instance) => <String, dynamic>{
      'name': instance.name,
      'gender': instance.gender,
      'age': instance.age,
    };
```
Is this expectation reasonable or there is a fundamental misunderstanding?

**Comment by mdeandrea-mrmilu** [comment]
I had the same error when extends a model with another class
This is a bug

**Comment by kevmoo** [comment]
The current behavior is a feature (kinda). We ignore "getters" without setters (including fields) to make sure things round-trip.

If you set `createFactory: false` we just grab all of the fields.

Recommendation: just ignore the "getters" you don't want with https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/ignore.html

**Comment by ctrysbita** [comment]
@kevmoo but how to include getters when using createFactory: true ?

**Comment by angel333** [comment]
> @kevmoo but how to include getters when using createFactory: true ?

I guess creating an **empty setter** is the cleanest solution.

I was dealing with the same issue. Because of some Firestore limitations I needed to store a field with some aggregated data. I ended up creating a getter and an empty setter.

```
List<String> get aggregated =>... getter code...;
set aggregated(List<String> _) {} // empty
```

It makes sense to me that `fromJson` and `toJson` should indeed be consistent. Otherwise `disallow_unrecognized_keys` would break.
